server/data/data-api/src/main/java/org/apache/james/rrt/api/RecipientRewriteTable.java

```diff
@@ -106,6 +109,31 @@ public String toString() {
      */
     Map<MappingSource, Mappings> getAllMappings() throws RecipientRewriteTableException;
 
+    default List<MappingSource> listSources(Mapping mapping) throws RecipientRewriteTableException {
+        if (!isSupportedListSources(mapping)) {
```

Comment:
We should rather throw to make explicit the fact that it is not supported, no? (Preconditions)